### DISCUSSION TITLE
Unexpected (or is it) behaviour when originals are not available

### DISCUSSION THREAD

**Discussion by artincube** [photoprism/photoprism#3521] [General]
Hi, I just recently started testing Photoprism to sort and organize many TB of photographs and videos taken by half a dozen mobile phones (jpeg and raw), half a dozen dslrs and compacts, over the last 20 years or so.
I found the best performance with what I have at hand is with the Apple Mac Book Pro M1 Max 64GB on Ventura using Docker. The Photoprism folder is on the very fast internal storage and the originals on a USB C 3.2 gen 2 fast storage.

The idea is to be able to, in between sessions of importing and indexing, unplug the originals storage and carry on working on the metadata, editing and saving it in the Photoprism storage folder on the internal drive of the laptop through the web interface.

To Resume what is below:
If I don't have the originals available I cannot launch Photoprism and work on metadata.
If I unplug the originals while Photoprism is running I can carry on working on metadata and this is saved properly.
The behaviour I would be happy with is that Photoprism could still launch if originals are not available, warning the user in terminal and in the web UI that while originals cannot be downloaded or new items added to the library, metadata of all assets can be modified and saved.
Could Photoprism connect to the originals drive if it is plugged back in, without having to down/up?

For now, I am trying to figure out if what I am doing is "safe".

The volume for originals in the yml file is
      - "/Volumes/14-Photoprism/Originals:/photoprism/originals"
 
**If I try to compose up while the originals drive is not attached:**
I get an error and while mariadb gets running, photoprism does not launch.
Error response from daemon: error while creating mount source path '/host_mnt/Volumes/14-Photoprism/Originals': mkdir /host_mnt/Volumes/14-Photoprism: file exists

After I plug in the originals drive and compose up Photoprism again, it gives the same error:
Error response from daemon: error while creating mount source path '/host_mnt/Volumes/14-Photoprism/Originals': mkdir /host_mnt/Volumes/14-Photoprism: file exists
I have to restart Docker engine and then it works, no error, but it seems to me that sometimes it worked after a couple attempts without restarting the engine.

**If I unplug the originals drive after having started Photoprism properly:**
In the web UI,
Library>Originals : no pictures found, empty, but there is still the number of originals in the sidebar.
Search and others I can see all indexed media.
If I try to download something, I get files, dng and all, but they are 22 (zips) or 420 bytes (dng single).
That is unexpected, it should just tell me that originals are not available.
If I add metadata to a media, it saves properly and stays. That is good.

After I plug the originals drive back in while Photoprism is running nothing happens, I have to down and up Photoprism to see it again, then I see them in the Library>Originals in the web UI but they cannot be downloaded (this time it just tells me no file available instead of downloading 22-420 bytes files).
I do it again and now all seems to be as expected.

**Comment by lastzero** [comment]
An "offline mode" isn't implemented currently
As a result, it also wasn't tested by us. Feel free to create a feature request for this. Also note that we are currently on vacation, so answering questions will take longer than usual.

**Comment by artincube** [comment]
That was not long. Happy vacations. 
Got it. I do that.

In the wait, could you tell me if you expect any destructive behavior in unplugging the originals while photoprism is running? If not I can just live with it for now.

And if there are some test procedures that can be done on my side I am available.

**Reply by lastzero** [reply]
By default, files that are missing will be flagged as missing. That's reversed once the files are found again.
